### ISSUE TITLE
Sidekiq Enterprise localization

### ISSUE THREAD

**Issue by mperham** [mperham/sidekiq#3949]
Sidekiq Enterprise has localization strings in GEMROOT/web/locales/en.yml.  Non-English customers are welcome to translate that file into their native language so that the Web UI extensions look natural to them.  Here's Sidekiq Enterprise's en.yml:

```
en:
  Limits: Limits
  NoLimitsFound: No limits found
  Size: Size
  Name: Name
  Type: Type
  InUse: In Use
  Loop: Periodic Job
  Loops: Periodic Jobs
  NoLoopsFound: No periodic jobs found
  Schedule: Schedule
  Options: Options
  Held: Held
  HoldTime: Hold Time
  Waited: Waited
  WaitTime: Wait Time
  Overages: Overages
  Reclaimed: Reclaimed
```

Example UI screenshots:

https://github.com/mperham/sidekiq/wiki/Ent-Rate-Limiting#web-ui
https://github.com/mperham/sidekiq/wiki/Ent-Periodic-Jobs#web-ui

**Comment by hanahiroAze** [comment]
@mperham I'd like to translate, but how can I make pull request to pro/enterprise?
can I add these words to https://github.com/mperham/sidekiq/blob/master/web/locales/en.yml ?

**Comment by mperham** [comment]
Add a comment here with the strings.

**Comment by hanahiroAze** [comment]
here is my idea.

```
ja:
  Limits: 制限
  NoLimitsFound: 同時実行数制限が設定されていません
  Size: サイズ
  Name:  名前
  Type: 種類
  InUse: 使用中
  Loop: 定期実行
  Loops: 定期実行
  NoLoopsFound: 定期実行ジョブがありません
  Schedule: スケジュール
  Options: オプション
  Held: ロック回数
  HoldTime: ロック秒数
  Waited: ロック解放待ち回数
  WaitTime: ロック解放待ち時間
  Overages: タイムアウト超過回数
  Reclaimed: 再ロック要求回数
```

**Comment by mperham** [comment]
thank you, this will go in the next release

**Comment by gustavodiel** [comment]
Hello!

Here are the locations for Brazilian and European Portuguese:
```yaml
pt:
  Limits: Limites
  NoLimitsFound: Nenhum limite encontrado
  Size: Tamanho
  Name: Nome
  Type: Tipo
  InUse: Em uso
  Loop: Tarefa Periódica
  Loops: Tarefas Periódicas
  NoLoopsFound: Nenhuma tarefa periódica encontrada
  Schedule: Cronograma
  Options: Opções
  Held: Bloqueado
  HoldTime: Tempo Bloqueado
  Waited: A Esperar
  WaitTime: Tempo de Espera
  Overages: Invalidez por Idade
  Reclaimed: Bloqueios Recuperados
```

```yaml
pt-br:
  Limits: Limites
  NoLimitsFound: Nenhum limite encontrado
  Size: Tamanho
  Name: Nome
  Type: Tipo
  InUse: Em uso
  Loop: Tarefa Periódica
  Loops: Tarefas Periódicas
  NoLoopsFound: Nenhuma tarefa periódica encontrada
  Schedule: Cronograma
  Options: Opções
  Held: Bloqueado
  HoldTime: Tempo Bloqueado
  Waited: Esperando
  WaitTime: Tempo de Espera
  Overages: Invalidez por Idade
  Reclaimed: Bloqueios Recuperados
```

**Comment by redwarewolf** [comment]
A possible spanish translation.

```
es:
  Limits: Límites
  NoLimitsFound: Ningún Límite Encontrado
  Size: Tamaño
  Name: Nombre
  Type: Tipo
  InUse: En Uso
  Loop: Tarea Periodica
  Loops: Tareas Periodicas
  NoLoopsFound: Ninguna Tarea Periodica Encontrada
  Schedule: Cronograma
  Options: Opciones
  Held: Retenidos
  HoldTime: Tiempo Retenido
  Waited: Esperado
  WaitTime: Tiempo de Espera
  Overages: Exceso
```
  Reclaimed: Reclamado

**Comment by mperham** [comment]
Thank you @redwarewolf, these will be in the next Pro and Ent releases. 😎

**Comment by hanahiroAze** [comment]
@mperham 
The word `Immediate ` looks not translated.
here is my ja version screen.

![2019-03-01 14 09 53](https://user-images.githubusercontent.com/9178166/53617704-bf2a6e00-3c2b-11e9-84b4-226384634137.png)

**Comment by mperham** [comment]
@hakanensari Thank you, I missed this word in en.yml. I've updated the description. The "Immediate" metric shows "Number of times the caller got a concurrent lock immediately, without wait"

**Comment by gustavodiel** [comment]
@mperham do you need a update on the Portugues version with the `Immediate`?

**Comment by mperham** [comment]
Yep, if anyone wants to add Immediate here, I will update the locales so the next version is fixed.

**Comment by gustavodiel** [comment]
For both `pt` and `pt-br` it is:
```yaml
immediate: Imediatos
```

**Comment by mperham** [comment]
thank you!

**Comment by mperham** [comment]
@hakanensari Can you supply a translation so I can update JA and release?

**Comment by hanahiroAze** [comment]
@mperham 

maybe you're making the wrong mention...
it is ja version

```
immediate: 即時ロック取得回数
```

**Comment by fchatterji** [comment]
French:
en:
  Limits: Limites
  NoLimitsFound: Aucune limite trouvée
  Size: Taille
  Name: Nom
  Type: Type
  InUse: En cours d'utilisation
  Loop: Tâche périodique
  Loops: Tâches périodiques
  NoLoopsFound: Aucune tâche périodique trouvée
  Schedule: Programme
  Options: Options
  Held: Bloqué
  HoldTime: Temps de blocage
  Waited: En attente
  WaitTime: Temps d'attente
  Overages: Excédents
  Reclaimed: Récupéré
  Immediate: Immédiat

**Comment by mperham** [comment]
Merci, @fchatterji!

**Comment by loed-idzinga** [comment]
I've made an attempt to translate these terms into Dutch:

```
nl:
  Limits: Limieten
  NoLimitsFound: Geen limieten gevonden
  Size: Grootte
  Name: Naam
  Type: Type
  InUse: In gebruik
  Loop: Periodieke taak
  Loops: Periodieke taken
  NoLoopsFound: Geen periodieke taken gevonden
  Schedule: Tijdsindeling
  Options: Opties
  Held: Vastgehouden
  HoldTime: Tijd vastgehouden
  Waited: In afwachting
  WaitTime: Wachttijd
  Overages: Overschot
  Reclaimed: Teruggewonnen
  Immediate: Onmiddelijk
```

**Comment by davidmetta** [comment]
Italian version:

```YML
it:
  Limits: Limiti
  NoLimitsFound: Nessun Limite trovato
  Size: Quantità
  Name: Nome
  Type: Tipo
  InUse: In Uso
  Loop: Lavoro Periodico
  Loops: Lavori Periodici
  NoLoopsFound: Nessuno Lavoro Periodico trovato
  Schedule: Programma
  Options: Opzioni
  Held: Tenuti
  HoldTime: Tempo di tenuta
  Waited: Attesi
  WaitTime: Tempo di attesa
  Overages: Eccedenze
  Reclaimed: Reclamati
  Immediate: Immediati
```

**Comment by mobilutz** [comment]
Here is a german translation
```
de:
  Limits: Grenzen
  NoLimitsFound: Keine Grenzen gefunden
  Size: Größe
  Name: Name
  Type: Typ
  InUse: In Bearbeitung
  Loop: Periodischer Job
  Loops: Periodische Jobs
  NoLoopsFound: Keine periodischen Jobs gefunden
  Schedule: Pläne
  Options: Optionen
  Held: Einbehalten
  HoldTime: Einbehaltezeit
  Waited: Wartend
  WaitTime: Wartezeit
  Overages: Veraltete
  Reclaimed: Wiedergewonnen
  Immediate: Unmittelbar
```

**Comment by alekgosk** [comment]
polish translation:

```yml
pl:
  Limits: Limity
  NoLimitsFound: Nie znaleziono limitòw
  Size: Rozmiar
  Name: Nazwa
  Type: Typ
  InUse: W użyciu
  Loop: Okresowe zadanie
  Loops: Okresowe zadania
  NoLoopsFound: Nie znaleziono okresowych zadań
  Schedule: Harmonogram
  Options: Opcje
  Held: Zachowane
  HoldTime: Czas blokowania
  Waited: Czekanie
  WaitTime: Czas oczekiwania
  Overages: Nadwyżki
  Reclaimed: Odzyskano
  Immediate: Natychmiastowy
```

**Comment by mperham** [comment]
👍🏼😎

**Comment by ogom** [comment]
Added `Periodic` to Japanese

```yaml
ja:
  Limits: 制限
  NoLimitsFound: 同時実行数制限が設定されていません
  Size: サイズ
  Name: 名前
  Type: 種類
  InUse: 使用中
  Loop: 定期実行
  Loops: 定期実行
  Periodic: 定期実行
  NoLoopsFound: 定期実行ジョブがありません
  Schedule: スケジュール
  Options: オプション
  Held: ロック回数
  HoldTime: ロック秒数
  Waited: ロック解放待ち回数
  WaitTime: ロック解放待ち時間
  Overages: タイムアウト超過回数
  Reclaimed: 再ロック要求回数
  Immediate: 即時ロック取得回数
```

**Comment by gunnner** [comment]
Hello, here's Ukrainian verion:

```yaml
uk:
  Limits: Ліміти
  NoLimitsFound: Обмежень не знайдено
  Size: Розмір
  Name: Назва
  Type: Тип
  InUse: У використанні
  Loop: Періодичне завдання
  Loops: Періодичні завдання
  NoLoopsFound: Періодичних завдань не знайдено
  Schedule: Розклад
  Options: Параметри
  Held: Затримано
  HoldTime: Час затримки
  Waited: Очікувань
  WaitTime: Час очікування
  Overages: Перевищення
  Reclaimed: Звільнено
  Immediate: Без очікування
```

**Comment by mperham** [comment]
Thank you @gunnner!

**Comment by sobrinho** [comment]
@mperham How do you feel about bringing those files into the Sidekiq gem itself?

It can be tricky to handle them outside sidekiq-pro/sidekiq-ent, but it would make it easier for the community to help with them. We could keep the en.yml in pro/ent as a fallback in case the versions are mismatched.

Alternatively, it might be less exotic to create a separate sidekiq-i18n gem similar to https://github.com/svenfuchs/rails-i18n and https://github.com/devise-i18n/devise-i18n, which would avoid having files in the gem that it doesn't use.

_In other words, we would keep the en.yml on the gems and the i18n gem for other languages, similar to the way Rails and Devise does._

**Comment by mperham** [comment]
@sobrinho The issue is that strings can be added to any version and it would force sidekiq-ent to update its gemspec to require a new Sidekiq patch version that includes the string. It's been easier to just include the locale YML with the code that uses it.

All of that said, since the strings don't change very often anymore, maybe it would be ok to move the strings into Sidekiq OSS now.

**Comment by seuros** [comment]
IMO, dropping the built-in I18n would remove a lot of maintenance burden.

The Sidekiq dashboard is a developer tool. Its users are generally expected to have at least a basic understanding of English in 2026.

1. If the dashboard is in English, modern browsers can translate it locally with no network call. AI assistants in browsers can also translate the UI into virtually any language on demand.
2. Supporting RTL languages such as Arabic and Hebrew adds complexity to the UI and increases the chance of layout regressions and other side effects. if centering a div is hard. RTL layout that flip is next level. :)
3. The translations are incomplete and difficult to maintain. I once used AI to generate translations for every language supported by browsers. The resulting translation files ended up being larger than the Sidekiq codebase itself.

4. Supporting one language often creates expectations for related ones. For example, Russian and Ukrainian are different languages, so having a Russian translation doesn’t satisfy Ukrainian users. Arabic isnt a single universal translation in practice, as terminology differs across Arabic-speaking countries. You can see that Brazilians added their version of [Portuguese](https://github.com/sidekiq/sidekiq/blob/main/web/locales/pt-BR.yml). China/Taiwan...

In all the years I have used Sidekiq, I dont think I have seen a project seriously rely on the stock translations. Teams either build their own dashboard on top of the API or simply use the default English UI.

For Sidekiq 9.0, I propose in removing the built-in translations entirely. The existing translations could live in the wiki or website, and users who want localization could inject their own translation files through configuration.

**Comment by mperham** [comment]
If you consider services like Mastodon which use Sidekiq and expose its UI to the admin user, it becomes a little more reasonable to want broad language support. I want Sidekiq to be usable in many languages, the maintenance has been minimal in my experience.

**Comment by sobrinho** [comment]
@mperham Alternatively, you could create a public repo with the pro/ent translation files so the community can help.

All you would have to do is write some kind of script to pull those into the project during a new release — that's also feasible and doesn't require managing the gem itself:

```bash
curl -s https://api.github.com/repos/sidekiq/sidekiq-i18n/contents/web/locales \
  | grep -o '"download_url": "[^"]*\.yml"' \
  | cut -d'"' -f4 \
  | while read url; do
      curl -sL -o "web/locales/$(basename "$url")" "$url"
    done
```

**Comment by seuros** [comment]
I get it, and that’s why I proposed making translations configurable instead of shipping them with Sidekiq.

An admin who speaks only Italian or French gets a translated UI today, but someone using Hungarian, Latvian, or many other languages still falls back to English.

Every language added becomes part of the release and maintenance process. Translation updates end up requiring a Sidekiq release, and every release carries the risk that a native speaker says, "That’s not how we say it" or "This terminology is wrong in our region."

By moving translations out of the core,  users can update their locale with a script like @sobrinho's example..

**Comment by sobrinho** [comment]
My proposal is more for @mperham himself to run the script just before the release and copy the YAML files to the published gem.

So, everything remains as is — each gem keeps its own files instead of them living in Sidekiq OSS.

The advantage is that the community can see the YAML files and help with them, and @mperham would then run the script to copy the files instead of relying on this thread to copy them manually.

**Comment by seuros** [comment]
>  run the script just before the release and copy the YAML files to the published gem.

That still complexity.  Every change will require new version release.  He won't release if someone wants to add Esperanto someday.

Also the new update will bring whatever new feature/bug fix with it.. In some companies, code change requires re-audit by a technical senior member. Translations are handled by lower level staff.

**Comment by sobrinho** [comment]
That's already the cadence today; I don't see an issue with it.

What I'm suggesting is to open the files in a repo to be managed by the community, but updates would still follow the same release cadence. It's an improvement over the current process of managing this thread, compared to reviewing PRs and running a script prior to each release.

**Comment by mperham** [comment]
I'll likely just move the Pro/Ent strings into sidekiq's YML. There's really not a lot of them and they don't change very frequently so I don't see much need to create more infrastructure or process to manage.

**Comment by sobrinho** [comment]
Sounds good, @mperham!
